src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -1,23 +1,73 @@
 import _ from 'lodash/fp'
 import { useState } from 'react'
-import { div, h, label, span } from 'react-hyperscript-helpers'
+import { div, h, label, p, span } from 'react-hyperscript-helpers'
 import { ButtonPrimary, Link, spinnerOverlay } from 'src/components/common'
+import { warningBoxStyle } from 'src/components/data/data-utils'
+import { icon } from 'src/components/icons'
 import { TextInput } from 'src/components/input'
 import Modal from 'src/components/Modal'
 import { Ajax } from 'src/libs/ajax'
-import { bucketBrowserUrl } from 'src/libs/auth'
+import { bucketBrowserUrl, getUser } from 'src/libs/auth'
+import colors from 'src/libs/colors'
 import { reportError } from 'src/libs/error'
+import { isAppDeletable } from 'src/libs/runtime-utils'
 import * as Utils from 'src/libs/utils'
 
 
 const DeleteWorkspaceModal = ({ workspace: { workspace: { namespace, name, bucketName } }, onDismiss, onSuccess }) => {
   const [deleting, setDeleting] = useState(false)
   const [deleteConfirmation, setDeleteConfirmation] = useState('')
+  const [loadingApps, setLoadingApps] = useState(true)
+
+  const LoadApps = workspaceName => {
```

Comment:
Is `LoadApps` a component within the `DeleteWorkspaceModal` component as I see it has its own state which it is having to return.

I'm curious why one would want to nest a component, rather than having that state at the top level and just having a function for the loadApps work ? It seems to work when testing on the UI I'm just not sure I've seen that pattern before

Reply:
Hmm.. I played around this a lot initially and I was having trouble setting the state up properly (and getting more familiar with hooks). React was complaining about not being able to use hooks in a function that doesn't define a functional component, so I modeled this after how we load apps/runtimes in the WorkspaceCoontainer https://github.com/DataBiosphere/terra-ui/blob/dev/src/pages/workspaces/workspace/WorkspaceContainer.js#L153. I can see the benefit in that you can split state logic into smaller components that care about it, and then pass that around (for example, multiple components care about refreshingRuntimes). WorkspaceContainer is a bit different in that the state its tracking is more complex and multiple components need to trigger the state update though. 

I took another stab at it, and I think I have it working in a plain function.